### PR TITLE
ME: duplicate local record

### PR DESCRIPTION
**Pull request by LHBruneton-C2C**
### Description

This PR introduces the local record duplication feature, from the action menu on each record, in all records tables.

The click interaction on a record to open it has been moved to only the title, to allow a click on the menu.

When clicking on the duplicate action, the app navigates to a new duplicate route, which take an UUID as parameter, to read a local record, save it as a copy draft, then enter edition mode.

### Architectural changes

The ui-search results-table component now depends on the MatMenuModule from Angular Material.

### Screenshots

image

### Quality Assurance Checklist

- [X] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [X] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

### CODE REVIEW

File: libs/api/repository/src/lib/gn4/gn4-repository.ts
Comment (jahow) [inline libs/api/repository/src/lib/gn4/gn4-repository.ts:243]: you can probably leave that aside and have everything in the same `switchMap` :)

File: libs/api/repository/src/lib/gn4/gn4-repository.ts
Comment (jahow) [inline libs/api/repository/src/lib/gn4/gn4-repository.ts:246]: The original XML document should be provided to `writeRecord` so that only the affected parts of the XML are changed, otherwise an completely new XML doc will be generated from scratch.

```suggestion
        const xml = await converter.writeRecord(record, xml)
```

File: libs/ui/search/src/lib/results-table/results-table.component.html
Comment (jahow) [inline libs/ui/search/src/lib/results-table/results-table.component.html:129]: do you think you could use a gn-ui-button there? to have some hover/focus effects (the stop propagation part will also be handled that way)

**Review by jahow — APPROVED** [review]
Looking and working great, thanks! I made a few suggestions in the code, especially when generating the XML of the duplicated record.

Also it looks like this PR is rebased on another one, but I'm not sure which. Should we wait for another PR to be merged before merging this one?

File: libs/ui/search/src/lib/results-table/results-table.component.html
Comment (LHBruneton-C2C) [reply libs/ui/search/src/lib/results-table/results-table.component.html:129]: I had tried, but I'll try again.
Opening the menu didn't seem to work with a gn-ui-button, maybe because of the stop propagation indeed.

**Pull request merged**
